packages/react-ui-core/src/Gmap/Marker.js

```diff
@@ -0,0 +1,63 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { MapContext } from './MapContext'
+
+const EVENTS = {
+  onClick: 'click',
+  onDoubleClick: 'dblclick',
+  onDragEnd: 'dragend',
+  onMouseDown: 'mousedown',
+  onMouseOut: 'mouseout',
+  onMouseOver: 'mouseover',
+  onMouseUp: 'mouseup',
+  onRecenter: 'recenter',
+}
+
+class Marker extends PureComponent {
+  componentWillUnmount() {
+    if (this.marker) this.marker.setMap(null)
+  }
+
+  setupMarker(map) {
+    if (map) {
+      this.marker = new window.google.maps.Marker({
+        ...this.props,
+        map,
+      })
+
+      this.setupEvents()
+    }
+
+    return null
+  }
+
+  setupEvents() {
+    Object.keys(EVENTS).forEach(name => {
+      if (this.props[name]) {
+        this.marker.addListener(EVENTS[name], this.handleEvent(name))
```

Comment:
do we need to worry about tearing these down on unload?

Reply:
Hmm... I thought about this and was thinking marker was getting removed on unmount, but it's only getting removed from the map on unmount, so good catch.